src/compiler/core.ts

```diff
@@ -1779,6 +1752,8 @@ export function createSet<TElement, THash = number>(getHashCode: (element: TElem
             return size;
         },
         forEach(action: (value: TElement, key: TElement, set: Set<TElement>) => void): void {
+            // NOTE: arrayFrom means that if the callback mutates the underlying collection,
+            //       we won't have an accurate set of values

```

Comment:
```suggestion
            // NOTE: arrayFrom prevents `action` from modifying the map while we iterate
```

Reply:
To be honest, I'm not sure if the copy was intentional.

Reply:
Nothing fails when it's removed, huh

Reply:
Neither behavior is really easy to make guarantees about without extra book-keeping.

Right now if you add a new value and it's added into an array under the hood, you *will* witness a new value if you're in the middle of iterating, but not if you've already finished iterating over all values with a given key.

On the other hand, if the multimap stores a single value, you'll never see any added values as you iterate.